testmodule/testContracts.nim

```diff
@@ -188,3 +188,51 @@ for url in ["ws://localhost:8545", "http://localhost:8545"]:
       await provider.mineBlocks(2) # two additional blocks
       let receipt = await confirming
       check receipt.blockNumber.isSome
+
+    test "can query last block event log":
+
+      let signer0 = provider.getSigner(accounts[0])
+
+      discard await token.connect(signer0).mint(accounts[0], 100.u256)
+      discard await token.connect(signer0).transfer(accounts[1], 50.u256)
+
+      let logs = await token.queryFilter(Transfer)
+
+      check eventually logs == @[
+        Transfer(sender: accounts[0], receiver: accounts[1], value: 50.u256)
+      ]
+
+    test "can query past event logs by specifying from and to blocks":
+
+      let signer0 = provider.getSigner(accounts[0])
+      let signer1 = provider.getSigner(accounts[1])
+
+      discard await token.connect(signer0).mint(accounts[0], 100.u256)
+      discard await token.connect(signer0).transfer(accounts[1], 50.u256)
+      discard await token.connect(signer1).transfer(accounts[2], 25.u256)
+
+      let currentBlock = await provider.getBlockNumber()
+      let logs = await token.queryFilter(Transfer,
+                                         BlockTag.init(currentBlock - 3),
```

Comment:
Wouldn't it be a more interesting test if you checked only 2 blocks back? To ensure that it takes into account the `from` block number?

Reply:
Yes, good idea. Updated.